share/spack/setup-env.sh

```diff
@@ -214,14 +214,11 @@ fi
 # Figure out where this file is.  Below code needs to be portable to
 # bash and zsh.
 #
-_sp_source_file="${BASH_SOURCE[0]}"  # Bash's location of last sourced file.
-if [ -z "$_sp_source_file" ]; then
-    _sp_source_file="$0:A"           # zsh way to do it
-    if [[ "$_sp_source_file" == *":A" ]]; then
-        # Not zsh either... bail out with plain old $0,
-        # which WILL NOT work if this is sourced indirectly.
-        _sp_source_file="$0"
-    fi
+_sp_source_file=${${BASH_SOURCE[0]:-${(%):-%N}}:-""} # Bash/Zsh's location of last sourced file
```

Comment:
This is pretty illegible... You should add a link to where it comes from and try to explain how it works in comments above.

Reply:
Alright, fixed.
```
_sp_source_file=${BASH_SOURCE[0]:-"${(%):-%N}"}:-""} # Bash/Zsh's location of last sourced file
```
Can you delete this PR so I can open a new one? And I'll explain what the line is doing.

Reply:
Alright, I think this fixes it:
```
_sp_source_file=${BASH_SOURCE[0]:-${(%):-%N}}
```